code/clue.js

```diff
@@ -10,50 +8,214 @@ const mrGreen = {
+}
+
+const library = {
+  name: 'library',
+  image: 'https://images.photowall.com/products/45534.jpg?h=699&q=85'
+}
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+const studyRoom = {
+  name: 'study room',
+  image: 'https://i.pinimg.com/originals/90/d8/a5/90d8a5f9b6db6df0a8751b49a073d438.jpg'
+}
 
 const suspects = [
   mrGreen,
-  mrsWhite
-  // ...  and the rest
+  mrsWhite,
+  mrMustard,
+  mrsPeacock,
+  mrPlum,
+  missScarlet
 ]
 
-const weapons = []
+const weapons = [
+  rope, 
+  revolver,
+  candlestick,
+  knife, 
+  horseshoe,
+  poison
+]
 
-const rooms = []
+const rooms = [
+  hall, 
+  lounge, 
+  diningRoom,
+  kitchen,
+  ballRoom, 
+  conservatory, 
+  billardRoom, 
+  library, 
+  studyRoom
+]
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+let mystery = {
+  killer: '',
+  weapon: '',
+  room: ''
+}
 
-// This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
   mystery.killer = randomSelector(suspects)
-
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
   document.getElementById('killerCard').style.background = mystery.killer.color
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerOccupation').innerHTML = `${mystery.killer.occupation}`
+  document.getElementById('killerImage').src = mystery.killer.image
+  document.getElementById('killerDescription').innerHTML =`${mystery.killer.description}`
+}
+
+const pickWeapon = () => {
+  mystery.weapon =randomSelector(weapons)
+  document.getElementById('weaponCard').style.background = mystery.weapon.color
+  document.getElementById(
+    'weaponName'
+  ).innerHTML = `${mystery.weapon.name} ${mystery.weapon.weight} kg`
+  document.getElementById('killerWeapon').src = mystery.weapon.image
+}
+
+const pickRoom = () => {
+  mystery.room =randomSelector(rooms)
   document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+    'roomName'
+  ).innerHTML = `${mystery.room.name}`
+  document.getElementById('killerRoom').src = mystery.room.image
+}
+
+const revealMystery = () => {
+  let finalMessage = `The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room.name} with a ${mystery.weapon.name}.`
```

Comment:
This could be a `const` since you don't change its state!
```suggestion
  const finalMessage = `The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room.name} with a ${mystery.weapon.name}.`
```